eva/optimizer/optimizer_tasks.py

```diff
@@ -62,19 +53,17 @@ def task_type(self):
     def optimizer_context(self):
         return self._optimizer_context
 
-    @root_expr.setter
-    def root_expr(self, expr: GroupExpression):
-        self.root_expr = expr
-
     def execute(self):
         raise NotImplementedError
 
 
 class TopDownRewrite(OptimizerTask):
     def __init__(self, root_expr: GroupExpression,
+                 rule_set: List[Rule],
                  optimizer_context: OptimizerContext):
-        super().__init__(root_expr, root_expr.group_id,
-                         optimizer_context, OptimizerTaskType.TOP_DOWN_REWRITE)
+        self._root_expr = root_expr
+        self.rule_set = rule_set
+        super().__init__(optimizer_context, OptimizerTaskType.TOP_DOWN_REWRITE)
 
     @property
     def root_expr(self):
```

Comment:
Only TopDownRewrite uses @property and @setter. Shall we get rid of it for consistency?

Reply:
Sure, got it